onnxruntime/core/mlas/lib/quantize.cpp

```diff
@@ -422,3 +684,160 @@ Return Value:
 }
 
 #endif
+
+
+void
+MLASCALL
+MlasQLinearAddS8Kernel(
+    const int8_t* InputA,
+    float ScaleA,
+    int8_t ZeroPointA,
+    const int8_t* InputB,
+    float ScaleB,
+    int8_t ZeroPointB,
+    float ScaleC,
+    int8_t ZeroPointC,
+    int8_t* OutputC,
+    size_t LengthA,
+    size_t LengthB
+    )
+{
+    MlasQLinearAddKernel<int8_t>(
+        InputA, ScaleA, ZeroPointA,
+        InputB, ScaleB, ZeroPointB,
+        ScaleC, ZeroPointC, OutputC,
+        LengthA, LengthB
+    );
+}
+
+void
+MLASCALL
+MlasQLinearAddU8Kernel(
+    const uint8_t* InputA,
+    float ScaleA,
+    uint8_t ZeroPointA,
+    const uint8_t* InputB,
+    float ScaleB,
+    uint8_t ZeroPointB,
+    float ScaleC,
+    uint8_t ZeroPointC,
+    uint8_t* OutputC,
+    size_t LengthA,
+    size_t LengthB
+    )
+{
+    MlasQLinearAddKernel<uint8_t>(
+        InputA, ScaleA, ZeroPointA,
+        InputB, ScaleB, ZeroPointB,
+        ScaleC, ZeroPointC, OutputC,
+        LengthA, LengthB
+    );
+}
+
+
+template <typename T>
+class MlasQLinearAddPlatformKernel
+{
+public:
+    typedef void (MLASCALL KernelFunction)(
+        const T* InputA,
+        float ScaleA,
+        T ZeroPointA,
+        const T* InputB,
+        float ScaleB,
+        T ZeroPointB,
+        float ScaleC,
+        T ZeroPointC,
+        T* OutputC,
+        size_t LengthA,
+        size_t LengthB
+        );
+
+    typedef KernelFunction* PKernelFunction;
+
+    static PKernelFunction GetKernel();
+};
+
+template <>
+MlasQLinearAddPlatformKernel<uint8_t>::PKernelFunction
+MlasQLinearAddPlatformKernel<uint8_t>::GetKernel()
+{
+    return MlasPlatform.QLinearAddU8Kernel;
```

Comment:
this probably won't build on non-x64. instead of a class template, is a template function enough that is instantiated for uint8_t/int8_t and calls either through MlasPlatform or the direct kernel depending on #ifdefs?

Reply:
I removed all the GetKernel related class and code which is introduced before as logic complex with parellel etc. Now main logic in the entrance function is clear.